src/rpc/net.cpp

```diff
@@ -293,6 +294,20 @@ static RPCHelpMan addnode()
         throw std::runtime_error(
             self.ToString());
     }
+    std::string connection_type_name = "manual";
+    if (!request.params[2].isNull()) {
+        connection_type_name = request.params[2].get_str();
+    }
+    if (connection_type_name != "manual" && connection_type_name != "manual-block-relay") {
```

Comment:
Is it possible to use `ConnectionTypeAsString()` here to avoid defining these strings in multiple places in the source?

Reply:
Good idea! Used `ConnectionTypeAsString()` here and in the last commit (also rearranged the rpc processing code a bit).